test_failure_message_with_no_name fails with 3.11.7/3.12.1

I'm just upgrading nixpkgs to Python 3.11.7 and 3.12.1, and we're seeing `test_failure_message_with_no_name` fail on pytest-mock 3.12.0

```patch
-expected call not found.\nExpected: mock()\n  Actual: not called.
+expected call not found.\nExpected: mock()\nActual: not called.
```
Looks like a whitespace went missing.

<details><summary>Full log</summary>
<pre>
pytest-mock> _______________ TestMockerStub.test_failure_message_with_no_name _______________
pytest-mock> 
pytest-mock> self = <test_pytest_mock.TestMockerStub object at 0x7ffff5bdcf50>
pytest-mock> mocker = <pytest_mock.plugin.MockerFixture object at 0x7ffff5b94810>
pytest-mock> 
pytest-mock>     def test_failure_message_with_no_name(self, mocker: MagicMock) -> None:
pytest-mock> >       self.__test_failure_message(mocker)
pytest-mock> 
pytest-mock> tests/test_pytest_mock.py:254: 
pytest-mock> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pytest-mock> 
pytest-mock> self = <test_pytest_mock.TestMockerStub object at 0x7ffff5bdcf50>
pytest-mock> mocker = <pytest_mock.plugin.MockerFixture object at 0x7ffff5b94810>
pytest-mock> kwargs = {}, expected_name = 'mock'
pytest-mock> msg = 'expected call not found.\nExpected: {0}()\nActual: not called.'
pytest-mock> expected_message = 'expected call not found.\nExpected: mock()\nActual: not called.'
pytest-mock> stub = <MagicMock spec='function' id='140737315949840'>
pytest-mock> exc_info = <ExceptionInfo AssertionError('expected call not found.\nExpected: mock()\n  Actual: not called.') tblen=3>
pytest-mock> @py_assert2 = AssertionError('expected call not found.\nExpected: mock()\n  Actual: not called.')
pytest-mock> @py_assert4 = 'expected call not found.\nExpected: mock()\n  Actual: not called.'
pytest-mock> 
pytest-mock>     def __test_failure_message(self, mocker: MockerFixture, **kwargs: Any) -> None:
pytest-mock>         expected_name = kwargs.get("name") or "mock"
pytest-mock>         if NEW_FORMATTING:
pytest-mock>             msg = "expected call not found.\nExpected: {0}()\nActual: not called."
pytest-mock>         else:
pytest-mock>             msg = "Expected call: {0}()\nNot called"
pytest-mock>         expected_message = msg.format(expected_name)
pytest-mock>         stub = mocker.stub(**kwargs)
pytest-mock>         with pytest.raises(AssertionError) as exc_info:
pytest-mock>             stub.assert_called_with()
pytest-mock> >       assert str(exc_info.value) == expected_message
pytest-mock> E       AssertionError: assert 'expected call not found.\nExpected: mock()\n  Actual: not called.' == 'expected call not found.\nExpected: mock()\nActual: not called.'
pytest-mock> E           expected call not found.
pytest-mock> E           Expected: mock()
pytest-mock> E         - Actual: not called.
pytest-mock> E         +   Actual: not called.
pytest-mock> E         ? ++
pytest-mock> 
pytest-mock> tests/test_pytest_mock.py:251: AssertionError
pytest-mock> _____________ TestMockerStub.test_failure_message_with_name[None] ______________
pytest-mock> 
pytest-mock> self = <test_pytest_mock.TestMockerStub object at 0x7ffff5bde6d0>
pytest-mock> mocker = <pytest_mock.plugin.MockerFixture object at 0x7ffff5b5c3d0>
pytest-mock> name = None
pytest-mock> 
pytest-mock>     @pytest.mark.parametrize("name", (None, "", "f", "The Castle of aaarrrrggh"))
pytest-mock>     def test_failure_message_with_name(self, mocker: MagicMock, name: str) -> None:
pytest-mock> >       self.__test_failure_message(mocker, name=name)
pytest-mock> 
pytest-mock> tests/test_pytest_mock.py:258: 
pytest-mock> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pytest-mock> 
pytest-mock> self = <test_pytest_mock.TestMockerStub object at 0x7ffff5bde6d0>
pytest-mock> mocker = <pytest_mock.plugin.MockerFixture object at 0x7ffff5b5c3d0>
pytest-mock> kwargs = {'name': None}, expected_name = 'mock'
pytest-mock> msg = 'expected call not found.\nExpected: {0}()\nActual: not called.'
pytest-mock> expected_message = 'expected call not found.\nExpected: mock()\nActual: not called.'
pytest-mock> stub = <MagicMock spec='function' id='140737315717904'>
pytest-mock> exc_info = <ExceptionInfo AssertionError('expected call not found.\nExpected: mock()\n  Actual: not called.') tblen=3>
pytest-mock> @py_assert2 = AssertionError('expected call not found.\nExpected: mock()\n  Actual: not called.')
pytest-mock> @py_assert4 = 'expected call not found.\nExpected: mock()\n  Actual: not called.'
pytest-mock> 
pytest-mock>     def __test_failure_message(self, mocker: MockerFixture, **kwargs: Any) -> None:
pytest-mock>         expected_name = kwargs.get("name") or "mock"
pytest-mock>         if NEW_FORMATTING:
pytest-mock>             msg = "expected call not found.\nExpected: {0}()\nActual: not called."
pytest-mock>         else:
pytest-mock>             msg = "Expected call: {0}()\nNot called"
pytest-mock>         expected_message = msg.format(expected_name)
pytest-mock>         stub = mocker.stub(**kwargs)
pytest-mock>         with pytest.raises(AssertionError) as exc_info:
pytest-mock>             stub.assert_called_with()
pytest-mock> >       assert str(exc_info.value) == expected_message
pytest-mock> E       AssertionError: assert 'expected call not found.\nExpected: mock()\n  Actual: not called.' == 'expected call not found.\nExpected: mock()\nActual: not called.'
pytest-mock> E           expected call not found.
pytest-mock> E           Expected: mock()
pytest-mock> E         - Actual: not called.
pytest-mock> E         +   Actual: not called.
pytest-mock> E         ? ++
pytest-mock> 
pytest-mock> tests/test_pytest_mock.py:251: AssertionError
pytest-mock> _______________ TestMockerStub.test_failure_message_with_name[] ________________
pytest-mock> 
pytest-mock> self = <test_pytest_mock.TestMockerStub object at 0x7ffff5bdf290>
pytest-mock> mocker = <pytest_mock.plugin.MockerFixture object at 0x7ffff5f7bb50>, name = ''
pytest-mock> 
pytest-mock>     @pytest.mark.parametrize("name", (None, "", "f", "The Castle of aaarrrrggh"))
pytest-mock>     def test_failure_message_with_name(self, mocker: MagicMock, name: str) -> None:
pytest-mock> >       self.__test_failure_message(mocker, name=name)
pytest-mock> 
pytest-mock> tests/test_pytest_mock.py:258: 
pytest-mock> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pytest-mock> 
pytest-mock> self = <test_pytest_mock.TestMockerStub object at 0x7ffff5bdf290>
pytest-mock> mocker = <pytest_mock.plugin.MockerFixture object at 0x7ffff5f7bb50>
pytest-mock> kwargs = {'name': ''}, expected_name = 'mock'
pytest-mock> msg = 'expected call not found.\nExpected: {0}()\nActual: not called.'
pytest-mock> expected_message = 'expected call not found.\nExpected: mock()\nActual: not called.'
pytest-mock> stub = <MagicMock spec='function' id='140737320039632'>
pytest-mock> exc_info = <ExceptionInfo AssertionError('expected call not found.\nExpected: mock()\n  Actual: not called.') tblen=3>
pytest-mock> @py_assert2 = AssertionError('expected call not found.\nExpected: mock()\n  Actual: not called.')
pytest-mock> @py_assert4 = 'expected call not found.\nExpected: mock()\n  Actual: not called.'
pytest-mock> 
pytest-mock>     def __test_failure_message(self, mocker: MockerFixture, **kwargs: Any) -> None:
pytest-mock>         expected_name = kwargs.get("name") or "mock"
pytest-mock>         if NEW_FORMATTING:
pytest-mock>             msg = "expected call not found.\nExpected: {0}()\nActual: not called."
pytest-mock>         else:
pytest-mock>             msg = "Expected call: {0}()\nNot called"
pytest-mock>         expected_message = msg.format(expected_name)
pytest-mock>         stub = mocker.stub(**kwargs)
pytest-mock>         with pytest.raises(AssertionError) as exc_info:
pytest-mock>             stub.assert_called_with()
pytest-mock> >       assert str(exc_info.value) == expected_message
pytest-mock> E       AssertionError: assert 'expected call not found.\nExpected: mock()\n  Actual: not called.' == 'expected call not found.\nExpected: mock()\nActual: not called.'
pytest-mock> E           expected call not found.
pytest-mock> E           Expected: mock()
pytest-mock> E         - Actual: not called.
pytest-mock> E         +   Actual: not called.
pytest-mock> E         ? ++
pytest-mock> 
pytest-mock> tests/test_pytest_mock.py:251: AssertionError
pytest-mock> _______________ TestMockerStub.test_failure_message_with_name[f] _______________
pytest-mock> 
pytest-mock> self = <test_pytest_mock.TestMockerStub object at 0x7ffff5bdee10>
pytest-mock> mocker = <pytest_mock.plugin.MockerFixture object at 0x7ffff5cfa010>, name = 'f'
pytest-mock> 
pytest-mock>     @pytest.mark.parametrize("name", (None, "", "f", "The Castle of aaarrrrggh"))
pytest-mock>     def test_failure_message_with_name(self, mocker: MagicMock, name: str) -> None:
pytest-mock> >       self.__test_failure_message(mocker, name=name)
pytest-mock> 
pytest-mock> tests/test_pytest_mock.py:258: 
pytest-mock> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pytest-mock> 
pytest-mock> self = <test_pytest_mock.TestMockerStub object at 0x7ffff5bdee10>
pytest-mock> mocker = <pytest_mock.plugin.MockerFixture object at 0x7ffff5cfa010>
pytest-mock> kwargs = {'name': 'f'}, expected_name = 'f'
pytest-mock> msg = 'expected call not found.\nExpected: {0}()\nActual: not called.'
pytest-mock> expected_message = 'expected call not found.\nExpected: f()\nActual: not called.'
pytest-mock> stub = <MagicMock name='f' spec='function' id='140737317406032'>
pytest-mock> exc_info = <ExceptionInfo AssertionError('expected call not found.\nExpected: f()\n  Actual: not called.') tblen=3>
pytest-mock> @py_assert2 = AssertionError('expected call not found.\nExpected: f()\n  Actual: not called.')
pytest-mock> @py_assert4 = 'expected call not found.\nExpected: f()\n  Actual: not called.'
pytest-mock> 
pytest-mock>     def __test_failure_message(self, mocker: MockerFixture, **kwargs: Any) -> None:
pytest-mock>         expected_name = kwargs.get("name") or "mock"
pytest-mock>         if NEW_FORMATTING:
pytest-mock>             msg = "expected call not found.\nExpected: {0}()\nActual: not called."
pytest-mock>         else:
pytest-mock>             msg = "Expected call: {0}()\nNot called"
pytest-mock>         expected_message = msg.format(expected_name)
pytest-mock>         stub = mocker.stub(**kwargs)
pytest-mock>         with pytest.raises(AssertionError) as exc_info:
pytest-mock>             stub.assert_called_with()
pytest-mock> >       assert str(exc_info.value) == expected_message
pytest-mock> E       AssertionError: assert 'expected call not found.\nExpected: f()\n  Actual: not called.' == 'expected call not found.\nExpected: f()\nActual: not called.'
pytest-mock> E           expected call not found.
pytest-mock> E           Expected: f()
pytest-mock> E         - Actual: not called.
pytest-mock> E         +   Actual: not called.
pytest-mock> E         ? ++
pytest-mock> 
pytest-mock> tests/test_pytest_mock.py:251: AssertionError
pytest-mock> ___ TestMockerStub.test_failure_message_with_name[The Castle of aaarrrrggh] ____
pytest-mock> 
pytest-mock> self = <test_pytest_mock.TestMockerStub object at 0x7ffff5bdf5d0>
pytest-mock> mocker = <pytest_mock.plugin.MockerFixture object at 0x7ffff5cc9710>
pytest-mock> name = 'The Castle of aaarrrrggh'
pytest-mock> 
pytest-mock>     @pytest.mark.parametrize("name", (None, "", "f", "The Castle of aaarrrrggh"))
pytest-mock>     def test_failure_message_with_name(self, mocker: MagicMock, name: str) -> None:
pytest-mock> >       self.__test_failure_message(mocker, name=name)
pytest-mock> 
pytest-mock> tests/test_pytest_mock.py:258: 
pytest-mock> _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pytest-mock> 
pytest-mock> self = <test_pytest_mock.TestMockerStub object at 0x7ffff5bdf5d0>
pytest-mock> mocker = <pytest_mock.plugin.MockerFixture object at 0x7ffff5cc9710>
pytest-mock> kwargs = {'name': 'The Castle of aaarrrrggh'}
pytest-mock> expected_name = 'The Castle of aaarrrrggh'
pytest-mock> msg = 'expected call not found.\nExpected: {0}()\nActual: not called.'
pytest-mock> expected_message = 'expected call not found.\nExpected: The Castle of aaarrrrggh()\nActual: not called.'
pytest-mock> stub = <MagicMock name='The Castle of aaarrrrggh' spec='function' id='140737317209680'>
pytest-mock> exc_info = <ExceptionInfo AssertionError('expected call not found.\nExpected: The Castle of aaarrrrggh()\n  Actual: not called.') tblen=3>
pytest-mock> @py_assert2 = AssertionError('expected call not found.\nExpected: The Castle of aaarrrrggh()\n  Actual: not called.')
pytest-mock> @py_assert4 = 'expected call not found.\nExpected: The Castle of aaarrrrggh()\n  Actual: not called.'
pytest-mock> 
pytest-mock>     def __test_failure_message(self, mocker: MockerFixture, **kwargs: Any) -> None:
pytest-mock>         expected_name = kwargs.get("name") or "mock"
pytest-mock>         if NEW_FORMATTING:
pytest-mock>             msg = "expected call not found.\nExpected: {0}()\nActual: not called."
pytest-mock>         else:
pytest-mock>             msg = "Expected call: {0}()\nNot called"
pytest-mock>         expected_message = msg.format(expected_name)
pytest-mock>         stub = mocker.stub(**kwargs)
pytest-mock>         with pytest.raises(AssertionError) as exc_info:
pytest-mock>             stub.assert_called_with()
pytest-mock> >       assert str(exc_info.value) == expected_message
pytest-mock> E       AssertionError: assert 'expected call not found.\nExpected: The Castle of aaarrrrggh()\n  Actual: not called.' == 'expected call not found.\nExpected: The Castle of aaarrrrggh()\nActual: not called.'
pytest-mock> E           expected call not found.
pytest-mock> E           Expected: The Castle of aaarrrrggh()
pytest-mock> E         - Actual: not called.
pytest-mock> E         +   Actual: not called.
pytest-mock> E         ? ++
pytest-mock> 
pytest-mock> tests/test_pytest_mock.py:251: AssertionError
</pre>
</details>

--- a/tests/test_pytest_mock.py
+++ b/tests/test_pytest_mock.py
@@ -25,6 +25,8 @@ skip_pypy = pytest.mark.skipif(
 
 # Python 3.8 changed the output formatting (bpo-35500), which has been ported to mock 3.0
 NEW_FORMATTING = sys.version_info >= (3, 8)
+# Python 3.11.7 changed the output formatting, https://github.com/python/cpython/issues/111019
+NEWEST_FORMATTING = sys.version_info >= (3, 11, 7)
 
 if sys.version_info[:2] >= (3, 8):
     from unittest.mock import AsyncMock
@@ -240,7 +242,9 @@ class TestMockerStub:
 
     def __test_failure_message(self, mocker: MockerFixture, **kwargs: Any) -> None:
         expected_name = kwargs.get("name") or "mock"
-        if NEW_FORMATTING:
+        if NEWEST_FORMATTING:
+            msg = "expected call not found.\nExpected: {0}()\n  Actual: not called."
+        elif NEW_FORMATTING:
             msg = "expected call not found.\nExpected: {0}()\nActual: not called."
         else:
             msg = "Expected call: {0}()\nNot called"

